### DISCUSSION TITLE
Reading uploaded file

### DISCUSSION THREAD

**Discussion by kuzeykoca** [QuorumDMS/ftp-srv#284] [General]
connection.on('STOR', (error, fileName) => { // upload
        const text = ???
        console.log(text)
})

1 Is there any way to read incoming file at this step without instantiating a new fs object ?
2 How can i decide whether save the incoming file ?

**Comment by matt-forster** [comment]
Yep! Absolutely. 

You'll want to write your own `write` function and provide a custom filesystem. You do not have to provide every function - if they are missing they will use the default functions.

https://github.com/autovance/ftp-srv#writefilename-append-start

Here is the default implementation: https://github.com/autovance/ftp-srv/blob/master/src/fs.js#L79

For example, we use this write function to store our files in S3 and expose a purely virtual file system to the client.
